lib/sequel_tools.rb

```diff
@@ -40,7 +57,7 @@ def self.base_config(extra_config = {})
   def self.inject_rake_tasks(config = {}, rake_context)
     require_relative 'sequel_tools/actions_manager'
     require_relative 'sequel_tools/all_actions'
-    actions_manager = ActionsManager.new config
+    actions_manager = ActionsManager.new base_config(config)
```

Comment:
Hi @janko. The main purpose of calling base_config here is not really about sharing configs for different environments. I can see that this name is misleading and could be improved. But base_config does 3 things:

1 - set the default options when not specified;
2 - ensure all config params are actually valid to avoid people spending a lot of time trying to understand why it is not working as expected because they mistyped some option.
3 - it expands the relative paths to migrations_location, schema_location and seeds_location.

I don't really understand the reasoning behind this commit. Why removing base_config would help you setting up your projects? I mean, why troubles calling base_config cause to your projects?

Reply:
The main benefit is smaller API surface to remember. When I first saw `base_config`, I thought it created something that resembled a Hash but isn't a Hash, which increased complexity in my mind. All of the things you describe can also happen in `inject_rake_tasks`, and it's how most libraries work (e.g. Shrine's storage classes set default values and assert on required arguments directly in `#initialize`).

Consider the following example:

```rb
base_config = SequelTools.base_config(dbadapter: "postgresql", dbname: "myapp_development", username: "")

namespace "db" do
  SequelTools.inject_rake_tasks base_config.merge(log_level: :info, sql_log_level: :info), self
end

namespace "db:test" do
  SequelTools.inject_rake_tasks base_config.merge(dbname: "myapp_test"), self
end
```

Here I needed to set `:dbname` in base config, even though it's going to differ from test database, which I don't find intuitive. Not needing to use `base_config` allows the user to share only the common config if they wanted to, and you still get all those checks & conversion of arguments:

```rb
base_config = { dbadapter: "postgresql" }

namespace "db" do
  SequelTools.inject_rake_tasks base_config.merge(dbname: "myapp_development", log_level: :info, sql_log_level: :info), self
end

namespace "db:test" do
  SequelTools.inject_rake_tasks base_config.merge(dbname: "myapp_test", log_level: :info, sql_log_level: :info), self
end
```

And in case of single set of rake tasks, you get to avoid defining a local variable, and have all in a single call. As a Rubyist I like this; a lot of Ruby's standard library helps avoid defining local variables. So, while it doesn't cause any troubles, it does gives me more freedom.